### PR TITLE
Refactor `seq_file_parser.py` for performance and readability

### PR DESCRIPTION
**Pull request by ThibFrgsGmz**
| | |
|:---|:---|
|**_Originating Project/Creator_**| |
|**_Affected Component_**|  |
|**_Affected Architectures(s)_**|  |
|**_Related Issue(s)_**| void |
|**_Has Unit Tests (y/n)_**| n |
|**_Builds Without Errors (y/n)_**| Let CI run |
|**_Unit Tests Pass (y/n)_**| Let CI run |
|**_Documentation Included (y/n)_**| n |

---
## Change Description

This PR aims to:
- Remove the unnecessary else statement to limit nested statements (`def parseAbsolute` L210)
- Convert a `for` loop into a `list` comprehension and directlt returning it (`def replaceSpacesAndCommas` L69)
- Use f-string instead of string interpolation
- Replace `dict()` / `list()` with `{}` / `[]` for performance speed

## Rationale

Clean code, better performance and readability.

## Testing/Review Recommendations

void

## Future Work

void

### CODE REVIEW

File: src/fprime_gds/common/parsers/seq_file_parser.py [hunk 1]
```diff
@@ -85,25 +82,24 @@ def parseArg(arg):
...
-                            # See if it translates to a float:
-                            return float(arg)
-                        except ValueError:
-                            # Otherwise it is an enum type:
-                            return str(arg)
+                        # Otherwise it is an enum type:
+                        return str(arg)
 
-            return list(map(parseArg, args))
+            return [map(parseArg, args)]
```
Comment (LeStarch) [inline src/fprime_gds/common/parsers/seq_file_parser.py]: I think this line is wrong.  This creates a list of one item which is the object returned by the `map` call.

list(map(...)) instead iterates the result of the map call and makes it a `list` type.  It is equivalent to:

```
[item for item in map(...)]
```

Comment (ThibFrgsGmz) [reply src/fprime_gds/common/parsers/seq_file_parser.py]: Done!

**Review by LeStarch — APPROVED** [review]
Fixes review notes.  I am curious why you use a list comprehension as opposed to the original `list(map())` call as it is longer, but the inheritance issue has been fixed.

**Comment by ThibFrgsGmz** [comment]
> @LeStarch I am curious why you use a list comprehension as opposed to the original list(map()) call as it is longer

Since I learned that it is slower to use `dict`/`list` keywords, I tend to always want to delete them.

**Review by LeStarch — APPROVED** [review]

**Pull request merged**